osu.Game/Screens/Play/HUD/ArgonHealthDisplay.cs

```diff
@@ -0,0 +1,284 @@
+                new PathControlPoint(new Vector2(curve_end, bar_height)),
+                new PathControlPoint(new Vector2(curve_end + curve_smoothness, bar_height), PathType.Linear),
+                new PathControlPoint(new Vector2(bar_length, bar_height)),
+            });
+
+            var vertices = new List<Vector2>();
+            barPath.GetPathToProgress(vertices, 0.0, 1.0);
+
+            InternalChild = new FillFlowContainer
+            {
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Horizontal,
+                Spacing = new Vector2(4f, 0f),
+                Children = new Drawable[]
+                {
+                    new Circle
+                    {
+                        Margin = new MarginPadding { Top = 10f - 3f / 2f, Left = -2f },
+                        Size = new Vector2(50f, 3f),
+                    },
+                    new Container
+                    {
+                        AutoSizeAxes = Axes.Both,
+                        Children = new Drawable[]
+                        {
+                            new BackgroundPath
+                            {
+                                PathRadius = 10f,
+                                Vertices = vertices,
+                            },
+                            missBar = new BarPath
+                            {
+                                AutoSizeAxes = Axes.None,
+                                RelativeSizeAxes = Axes.Both,
+                                BarColour = miss_bar_colour,
+                                GlowColour = miss_bar_glow_colour,
+                                Alpha = 0f,
+                                PathRadius = 10f,
+                                Vertices = vertices
+                            },
+                            healthBar = new BarPath
+                            {
+                                AutoSizeAxes = Axes.None,
+                                RelativeSizeAxes = Axes.Both,
+                                BarColour = health_bar_colour,
+                                GlowColour = health_bar_glow_colour,
+                                PathRadius = 10f,
+                                Vertices = vertices
+                            },
+                        }
+                    }
+                },
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Current.BindValueChanged(v =>
+            {
+                if (v.NewValue > MissBarValue)
+                {
+                    missBar.FadeOut(300, Easing.OutQuint);
+                    resetMissBarDelegate?.Cancel();
+                    resetMissBarDelegate = null;
+                }
+
+                if (v.NewValue == 0)
+                    healthBar.FadeOut(300, Easing.OutQuint);
+                else if (healthBar.Alpha < 1)
+                    healthBar.FadeIn(300, Easing.OutQuint);
```

Comment:
This is wrong. It will cause a slower fade with more value changes. You probably want to move this to use a `ContinuousDamp` function in `Update` or similar.